lib/screens/search/product_search.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+
+import 'package:yisty_app/models/product.dart';
+import 'package:yisty_app/widgets/design/alert_page.dart';
+import 'package:yisty_app/services/product_service.dart';
+import 'package:yisty_app/services/rest_client/api_exceptions.dart';
+import 'package:yisty_app/widgets/design/empty_page.dart';
+import 'package:yisty_app/widgets/inherited_provider.dart';
+import 'package:yisty_app/widgets/products/product_list.dart';
+
+class ProductSearch extends SearchDelegate<Product> {
+  @override
+  List<Widget> buildActions(BuildContext context) {
+    if (query == '') {
+      return <Widget>[];
+    } else {
+      return <Widget>[
+        IconButton(icon: const Icon(Icons.close), onPressed: () {
+          query = '';
+        })
+      ];
+    }
```

Comment:
```suggestion
    if (query == '') {
      return <Widget>[];
    } 
    return <Widget>[
      IconButton(icon: const Icon(Icons.close), onPressed: () {
        query = '';
       })
     ];
    }
```
Lo dejaría así, ya que de igual manera si no se cumple la condición sale por el otro lado, también lo hace más legible segun el amigo code clean

Reply:
No tengo porque desconfiar del amigo code clean.